Replace the use of the function string.replace with the method str.replace

lparkes opened the pull request:
##### SUMMARY
Python 3 no longer has the module function `string.replace`, but it still shares the `str` method `replace` with Python 2. This PR replaces both uses of `string.replace()` in the Ansible source code with `str.replace()`. 

I also fixed a typo in a nearby comment.

##### ISSUE TYPE
- Bugfix Pull Request

##### COMPONENT NAME
service

##### ADDITIONAL INFORMATION
A couple of bits of code for managing BSD rc.conf style service configuration files still used the obsolete `string.replace()` function.

I have successfully tested this with the following trivial playbook
```
- hosts: fog
  tasks:
  - name: NTP service
    service:
      name: ntpd
      enabled: yes
```

lparkes committed:
Replace the use of the function string.replace() with the method str.replace()
because that's what works for both Python 2 and 3.

sivel reviewed (changes requested):
`import string` can also be removed.  Additionally, this should include a changelog fragment in `changelogs/fragments`

lparkes committed:
Cleanup the unused string import.

Added a changelog fragment.

lparkes commented:
Changes made.

lparkes commented:
Well, that rebase didn't work. The diff looked fine, so I'll dig into what git-fu I'm missing.

Akasurde commented:
@lparkes Could you please rebase this branch? You can use this guide - https://docs.ansible.com/ansible/latest/dev_guide/developing_rebasing.html

lparkes commented:
Thanks for pointing me at the documentation for the Github magic. This rebase has one fewer parents in the revision history and Github is sure to be happier with that. Please hold on while I double check that I have got all the Python 2 out of service.py. I (re)discovered another one in service.py over the weekend and I need double check that I've included it. It would be anoying to have to create another PR for the same sort of problem in the same source file.

lparkes committed:
The documentation for os.write() seems a bit iffy, but in Python 3 we
definitely cannot pass it a string and we need to encode it into
bytes.

The Python documentation at
https://docs.python.org/3/howto/pyporting.html#text-versus-binary-data
says that this code will work in Python 2 as well.

lparkes commented:
I added in a string to bytes conversion to keep os.write() happy. I would like to write a test for this, but a quick look shows me that this will require quite a bit of reading on my part.

lparkes commented:
I don't know of any remaining Python 2 code in the rcconf service support, so I'm happy for this to be merged.

lparkes commented:
I've rebased the PR again in order to trigger new CI tests.

bcoca commented:
@Iparkes  this is a good candidate for backporting, if you are interested https://docs.ansible.com/ansible/latest/community/development_process.html#backporting-merged-prs-in-ansible-base
